lib/inspec/control_eval_context.rb

```diff
@@ -92,6 +92,14 @@ def to_s
           res
         end
 
+        define_method :add_resource do |name, new_res|
```

Comment:
We can, but I'm not sure that will actually do what we want.  Since currently control code gets executed via instance_exec on this class, It'll still have access to private methods.